Basic setup

keren-elbaz-m opened the pull request:
Set up basic server and database connection with Express and Mongoose

keren-elbaz-m committed:
Basic setup

Set up basic server and database connection with Express and Mongoose

MoGuyda commented on src/server.ts:
```diff
@@ -0,0 +1,35 @@
+app.use(cookieParser());
+app.use(bodyParser.json());
+
+app.get('/health', (req, res) => {
```
Please create routes folder and inside a dedicated file for this route

keren-elbaz-m replied:
done

keren-elbaz-m committed:
health route

Refactored health check route into dedicated route file under routes folder

DanielBaronMoveo commented on src/server.ts:
```diff
@@ -0,0 +1,34 @@
+app.use(cookieParser());
+app.use(bodyParser.json());
+
+app.use('/', healthRoute);
```
types directory with enum dir

keren-elbaz-m replied:
done

keren-elbaz-m committed:
Create constants file for route paths

Created 'constants/routes.constants.ts' to store centralized route paths. and added ROOT and HEALTH path constants.
Updated `server.ts` to use the new constants

DanielBaronMoveo commented on src/routes/health.route.ts:
```diff
@@ -0,0 +1,10 @@
+
+const router = Router();
+
+router.get('/health', (req, res)=>{
```
Why don't we use the new routes constant object for the health route?

keren-elbaz-m replied:
done

DanielBaronMoveo commented on src/server.ts:
```diff
@@ -0,0 +1,33 @@
+
+app.use(ROUTES.HEALTH, healthRoute);
+
+const startServer = async()=>{
+    await connectDB();
+    const server = http.createServer(app);
+
+    server.listen(process.env.PORT || 8080, ()=>{
+    console.log(`Server running on http://localhost:${process.env.PORT}/`);
+});
+} 
```
Where are the try/catch blocks?

keren-elbaz-m replied:
done - added try and catch

DanielBaronMoveo commented on src/server.ts:
```diff
@@ -0,0 +1,33 @@
+}));
+
+app.use(cookieParser());
+app.use(bodyParser.json());
```
Let's get rid of body-parser since it's the old way of doing it. 
Since the latest versions we can use express.json() directly.

```
// Old way (still works, but redundant)
import bodyParser from 'body-parser';
app.use(bodyParser.json());

// New way (cleaner)
app.use(express.json());
```

If everything works for you as expected (test the call with Postman) leave it with express.json and remove the body-parser library.

keren-elbaz-m replied:
removed the body parser and updated to express.json (after testing).

DanielBaronMoveo reviewed (changes requested):
Minor changes, also watch a quick video of Postman and ofcourse install it. Pay attention to how you trigger API calls through this platform.

keren-elbaz-m committed:
Refactored server setup

Introduced RoutePath enum and updated ROUTES constant to use it.
Added a /health endpoint wrapped with try-catch.
Replaced body-parser with express.json() after testing.
